sdk/python/kfp/components/_components.py

```diff
@@ -457,6 +467,26 @@ def expand_command_part(arg) -> Union[str, List[str], None]:
 
             return output_filename
 
+        elif isinstance(arg, InputUriPlaceholder):
+            input_name = arg.input_uri_spec.input_name
+            input_argument = argument_values.get(input_name, None)
+            if input_name in argument_values:
+                input_uri = input_uri_generator(input_name)
+                input_uris[input_name] = input_uri
+                return input_uri
+            else:
+                input_spec = inputs_dict[input_name]
+                if input_spec.optional:
+                    return None
+                else:
+                    raise ValueError('No value provided for input {}'.format(input_name))
```

Comment:
nit: Perhaps we can use f-string here?

Reply:
We're still supporting Python 3.5.... We're going to drop support right after 1.1.